lib/beaker-hostgenerator/data.rb

```diff
@@ -29,6 +43,873 @@ def pe_upgrade_family
       'm' => 'master',
     }
 
+    BASE_CONFIG = {
+      'HOSTS' => {},
+      'CONFIG' => {
+        'nfs_server' => 'none',
+        'consoleport' => 443,
+      }
+    }
+
+    def base_host_config
+      {
+        'pe_dir' => pe_dir(pe_version, pe_family),
+        'pe_ver' => pe_version,
+        'pe_upgrade_dir' => pe_dir(pe_upgrade_version, pe_upgrade_family),
+        'pe_upgrade_ver' => pe_upgrade_version,
+      }
+    end
+
+    def osinfo
```

Comment:
Yeah, in my opinion part of the problem is probably somewhere in the `deep_merge` module, but also in the fact Ruby's module level "constants" are actually just references to totally mutable data structures. You basically have to explicitly create a deep copy of the data structure before using and manipulating it or you risk modifying a reference that will eventually be used elsewhere.